packages/react-charts/src/components/Patterns/examples/patterms.md

```diff
@@ -0,0 +1,791 @@
+        result.push(['pie']);
+      });
+      return result;
+    };
+
+    // Returns onMouseOver, onMouseOut, and onClick events for the interactive legend
+    this.getEvents = () => getInteractiveLegendEvents({
+      chartNames: this.getChartNames(),
+      isHidden: this.isHidden,
+      legendName: 'legend',
+      onLegendClick: this.handleLegendClick
+    });
+
+    // Returns legend data styled per hiddenSeries
+    this.getLegendData = () => {
+      const { hiddenSeries } = this.state;
+      return this.series.map((s, index) => {
+        return {
+          ...s.legendItem, // name property
+          ...getInteractiveLegendItemStyles(hiddenSeries.has(index)) // hidden styles
+        };
+      });
+    };
+
+    // Hide each data series individually
+    this.handleLegendClick = (props) => {
+      if (!this.state.hiddenSeries.delete(props.index)) {
+        this.state.hiddenSeries.add(props.index);
+      }
+      this.setState({ hiddenSeries: new Set(this.state.hiddenSeries) });
+    };
+
+    // Returns true if data series is hidden
+    this.isHidden = (index) => {
+      const { hiddenSeries } = this.state; // Skip if already hidden                
+      return hiddenSeries.has(index);
+    };
+
+    this.isDataAvailable = () => {
+      const { hiddenSeries } = this.state;
+      return hiddenSeries.size !== this.series.length;
+    };
+  };
+
+  render() {
+    const { hiddenSeries, width } = this.state;
+
+    const data = [];
+    this.series.map((s, index) => {
+      data.push(!hiddenSeries.has(index) ? s.datapoints : [{ y: null}]);
+    });
+
+    return (
+      <div style={{ height: '275px', width: '500px' }}>
+        <Chart
+          ariaDesc="Average number of pets"
+          ariaTitle="Pie chart example"
+          events={this.getEvents()}
+          height={275}
+          labels={({ datum }) => `${datum.x}: ${datum.y}`}
+          legendComponent={<ChartLegend name={'legend'} data={this.getLegendData()} />}
+          legendPosition="bottom"
+          padding={{
+            bottom: 65,
+            left: 20,
+            right: 20,
+            top: 20
+          }}
+          patternId="pattern_a" // Required for interactive legend functionality
+          showAxis={false}
+          themeColor={ChartThemeColor.multiUnordered}
+          usePatternDefs
+          width={500}
+        >
+          <ChartPie
+            constrainToVisibleArea={true}
+            data={data}
+            name="pie"
+          />
+        </Chart>
+      </div>
+    );
+  }
+}
+```
+
+### Interactive legend with area chart
+
+This demonstrates how to add an interactive legend using events such as `onMouseOver`, `onMouseOut`, and `onClick`.
+
+```js
+import React from 'react';
+import { 
+  Chart, 
+  ChartArea, 
+  ChartAxis, 
+  ChartGroup, 
+  ChartLegend,
+  ChartLegendTooltip,
+  ChartScatter, 
+  ChartThemeColor,
+  ChartVoronoiContainer,
```

Comment:
ChartVoronoiContainer is not used in this example.

Reply:
Done